Workhop instructions

rv0lt opened the pull request.

rv0lt committed:
workshop

rv0lt committed:
typos

aishling-scilifelab commented on README.md:
```diff
@@ -4,3 +4,9 @@ This repository contains the instructions for the hands-on sessions during the w
 
 The documentation for different sessions can be found in the `docs` directory. 
+
+Before starting, clone this repository in your computer:
```
I don't think we can assume that the attendees will have git installed and I think getting them to install git (especially if they are Windows users) will add a bunch of faff. 
Also, many won't have a nice markdown renderer installed locally but it will render nicely on GitHub 

Can we just ask them to download the files under data using the download button?

valyo replied:
We can mention both git and straight download (for those without git installed). Git is usually part of the reproducible research workflows and it will add an educational effect to our material, I believe.

rv0lt replied:
I rephrase it. 
But better if they download the whole repository as ZIP, if they download individual files, they can miss some of them and break the structure shown in the examples.

But anyway, it is not like it is asynchronous. We will be showing them in the screen this steps

aishling-scilifelab replied:
Totally get the point for the data directory (and maintaining that structure) but most of these people (likely) won't have an IDE with a markdown renderer so we still want them to look at the markdown instructions in GitHub (where it all looks nice and well-formatted). 

Could you zip the data files and commit that zipped file? Then they can simply download that.

rv0lt replied:
I mean yes. But they can already download as a zip the repo. 
And that doesn't imply they have to open the markdown files raw, like they can still follow the docs through the browser. I don't see a problem here.

aishling-scilifelab commented on data/example_directory_2/sub_directory_2/example_file_3.txt:
I love this example file 😆

aishling-scilifelab commented on docs/1_installation.md:
```diff
@@ -0,0 +1,36 @@
+
+The instructions for installation are available in our documentation, [accessible here.](https://scilifelabdatacentre.github.io/dds_cli/#install-the-command-line-interface-cli-dds-cli)
+
+They differ slightly between Mac or Windows. So, follow the section you need. Once installed, verify it by running:
```
Super minor but probably don't need that full stop/period after Windows. 

```suggestion
They differ slightly between Mac or Windows so follow the section you need. Once installed, verify it by running:
```

aishling-scilifelab commented on docs/1_installation.md:
```diff
@@ -0,0 +1,36 @@
+Run the help or version command again and verify that the output contains
+ `https://testing.delivery.scilifelab.se/` 
+and **not** 
+`https://delivery.scilifelab.se/`
```
Do we have any common issues during installation that we want to highlight the fix for?

valyo replied:
Joanna had an issue with missing `python3-devel ` package on Fedora linux, so I guess we can add that people with non Ubuntu/Debian based distros can possibly need to do smth like 
`sudo yum install python3-devel`

rv0lt replied:
What do you mean?

aishling-scilifelab replied:
As in, as well as assuming installation all goes well (especially as these users are supposed to support their data recipients in installing), it might be nice to talk about errors which can be encountered at this point (e.g. that have been previously reported to us).

rv0lt replied:
Weren't we asking them to have it already installed? I don't think we have written down a historical of previous reported cases. It can depend on a lot of things, I cannot think of concrete examples at this point

valyo replied:
That Fedora "issue" is the only one I can think of, and it is worth it mentioning it

rv0lt replied:
I have searched more deeply, and I cannot find any document talking about installation errors

valyo replied:
I don't think there's a document anywhere. This was Joanna trying to install the CLI on her Fedora computer and getting errors since Fedora doesn't seem to have that package installed by default. I don't think you'll find any other sources about such issue because there're no so many people around the world that tried to install our CLI and discussed the issue. The idea is that we mention it as a potential guide for people using other distros, since we cannot cover all of them, of cource.

aishling-scilifelab replied:
@valyo - can you summarise what Joanna had to install and I'll add some text in here?

aishling-scilifelab replied:
Actually, it still isn't working for Joanna so let's come back and add instructions for Fedora later.

aishling-scilifelab commented on docs/2_registration_project_access_session.md:
```diff
@@ -0,0 +1,142 @@
+
+Examples on how to use it are available in the [documentation.](https://scilifelabdatacentre.github.io/dds_cli/examples/#authentication-dds-auth)
+
+![enter image description here](https://scilifelabdatacentre.github.io/dds_cli/_images/dds-auth-login.svg)
```
The default accessibility alternate text still there - could add a brief description of the image.

aishling-scilifelab commented on docs/2_registration_project_access_session.md:
```diff
@@ -0,0 +1,142 @@
+
+![enter image description here](https://scilifelabdatacentre.github.io/dds_cli/_images/dds-auth-login.svg)
+
+ - [ ] TASK: Find the email and create an account, login filling your information. Note that the password will not being shown while you are typing it.
```
Grammar 

```suggestion
 - [ ] TASK: Find the email and create an account and login using your information. Note that the password will not be shown while you are typing it.
```

aishling-scilifelab commented on docs/2_registration_project_access_session.md:
```diff
@@ -0,0 +1,142 @@
+dds project create --title "<Project Title>" --description "<Project Description>" --principal-investigator "<Email to PI>"
+~~~
+
+> The email specified in the option `--principal-investigator` does not receive any emails; It’s only for information purposes at this time.
```
2 things: 
1) Might also be good to note that this doesn't automatically create an account for the PI (nor does it care whether a DDS a/c does exist with that address)
2) Capitalisation 
```suggestion
> The email specified in the option `--principal-investigator` does not receive any emails; it’s only for information purposes at this time.
```

aishling-scilifelab commented on docs/2_registration_project_access_session.md:
```diff
@@ -0,0 +1,142 @@
+[Docs.](https://scilifelabdatacentre.github.io/dds_cli/project/#dds-project-create)
+When the project is created, you should get an output similar to the one below. Remember the **Project ID**
+
+![enter image description here](https://scilifelabdatacentre.github.io/dds_cli/_images/dds-project-create.svg)
```
Accessible image description missing

aishling-scilifelab commented on docs/2_registration_project_access_session.md:
```diff
@@ -0,0 +1,142 @@
+
+![enter image description here](https://scilifelabdatacentre.github.io/dds_cli/_images/dds-project-create.svg)
+
+ - [ ] TASK: Run the create project command, remember to change the files of **title**, **description** and **pi**
```
Did we want to do anything here about the non-required options (researcher, owner and non-sensitive)?

I'm guessing there will be some people who have never used DDS but we will also have people who do already use it so might be good to try and make them aware of some options they don't already use.

rv0lt replied:
I think those more advanced questions, we can mention them live. Specially if they ask for them.

It will be difficult to account for all the specific options each command has, but all the commands explained have links to the docs, where all the flags are explained. And we can project the documentation while we explain the commands.

aishling-scilifelab replied:
I think if we're going to talk through it, we should have it written down here (possibly in a box or section labelled Optional). Given that there will be ~15 people, they likely won't remain perfectly in sync with each other and there will be people who won't be able to keep up with the live presentation aspect

rv0lt replied:
I will mention it as a note, okay

aishling-scilifelab commented on docs/2_registration_project_access_session.md:
```diff
@@ -0,0 +1,142 @@
+
+Check the folder which contains the data, inside this repository.
+~~~
+ls -R data/
```
This command won't work on Windows. 

Need to use `dir`

aishling-scilifelab commented on docs/2_registration_project_access_session.md:
```diff
@@ -0,0 +1,142 @@
+
+The general upload command is [`dds data put`.](https://scilifelabdatacentre.github.io/dds_cli/data/#dds-data-put)
+~~~
+dds data put --project "<Project ID>" --source "<File or directory to upload>"
```
Same comment as above: it might be cool to pick 1 or 2 of the options (especially if there are some that are more useful) and have users try them out too

rv0lt replied:
Some of the other flags from put are way more advanced.
I prefer to keep this tutorial simple. And during the session explain the documentation and the options.

valyo replied:
I agree with both of your points here, maybe we can have a task during the session where they look at the options and try to figure them out and/or raise questions about them

aishling-scilifelab replied:
Same as previous comment - realistically not all the attendees will be able to keep pace (and some will already be advanced users so will be familiar with these commands). If we are going to talk about the options then I think we should add a little section on that (but possibly labelling it optional)

rv0lt replied:
The optional flags for this command can be tricky, I can mention as a note that there are other options they can look up in the documentation (which they already have linked) but explain them can confuse more

aishling-scilifelab commented on docs/2_registration_project_access_session.md:
```diff
@@ -0,0 +1,142 @@
+
+# List contents
+
+You can interactively list the contents of a project with the [`ls` command:](https://scilifelabdatacentre.github.io/dds_cli/data/#dds-data-ls)
```
What does 'interactively' mean here?

rv0lt replied:
That the command prompts them again to specify a subdirectory to navigate through the files

aishling-scilifelab commented on docs/2_registration_project_access_session.md:
```diff
@@ -0,0 +1,142 @@
+dds data ls --project "<Project ID>"
+~~~
+
+However, You can view all files and directories in a project as a tree structure, using the `--tree` option.
```
```suggestion
However, you can view all files and directories in a project as a tree structure, using the `--tree` option.
```

aishling-scilifelab commented on docs/2_registration_project_access_session.md:
```diff
@@ -0,0 +1,142 @@
+~~~
+![enter image description here](https://scilifelabdatacentre.github.io/dds_cli/_images/dds-data-ls-tree.svg)
+
+ - [ ] TASK: Verify that all the data has being uploaded sucesfully
```
Typo 

```suggestion
 - [ ] TASK: Verify that all the data has being uploaded successfully
```

aishling-scilifelab commented on docs/2_registration_project_access_session.md:
```diff
@@ -0,0 +1,142 @@
+~~~
+![enter image description here](https://scilifelabdatacentre.github.io/dds_cli/_images/dds-data-ls-tree.svg)
+
+ - [ ] TASK: Verify that all the data has being uploaded sucesfully
```
Do we have any tips for users (possibly not DDS commands but Linux or Windows CMD commands) for sanity checking the upload given that some uploads might have 10,000s of files? 
For example, commands to count the number of files in a dir (e.g. `ls . | wc -l`) or get the total size of all files in the directory (e.g. `du -hs ./*`)

rv0lt replied:
Actually, you reminded me of something. 
One of these commands shows the size as Kibibytes, and our system stores the data as kilobytes (or the other way around, don't remember exactly). So when big uploads, there is a big difference on size if someone manually checks like that. Need to check it

aishling-scilifelab commented on docs/2_registration_project_access_session.md:
```diff
@@ -0,0 +1,142 @@
+# Release project and download data
+
+
+1. In order for researchers to get access to the data, you need to [change the status](https://scilifelabdatacentre.github.io/dds_cli/project/#dds-project-access) to `Available`.
```
Might be interesting to show them (even if we don't get them to test it) what researchers see if they try to download data when the project is not set to Available.

rv0lt replied:
It's not so mysterious, the client just stops and tells them to wait until the project is available (has happened a lot of times to me while locally testing).

Also, they get an email when it is released. So they know about it.

rv0lt replied:
I added a small explanation as a note

aishling-scilifelab commented on docs/2_registration_project_access_session.md:
```diff
@@ -0,0 +1,142 @@
+# Release project and download data
+
+
+1. In order for researchers to get access to the data, you need to [change the status](https://scilifelabdatacentre.github.io/dds_cli/project/#dds-project-access) to `Available`.
```
Might also be good to mention that putting the project into Available starts a countdown in number of days

rv0lt replied:
Should be mentioned in project life cycle

aishling-scilifelab commented on docs/2_registration_project_access_session.md:
```diff
@@ -0,0 +1,142 @@
+dds project access grant --project "<Project ID>" --email "<email>"
+~~~
+
+or
```
Might be good to note what the difference is: when should they pick one over the other?

rv0lt replied:
No difference at all, just two commands that do the same. The second one can be used to add any other use if they change the role. But for this purposes it is the same.

valyo replied:
Isn't the first command intended for existing accounts, i.e. it should fail if there is no account in the system with the specified email address (not sure if that's the behaviour, we should check it)

rv0lt replied:
Clarified that they do the same along with a reminder that they can also invite researcher when creating project

i-oden replied:
So I checked the code a bit quickly now and the `dds project access grant` command calls the `AddUser` resource / endpoint, which is the same one that `dds user add` calls, so there's no difference in functionality. I guess the idea is that they use `dds user add` (which was the only command from the beginning) as a default when they're inviting new accounts and that they use `dds project access grant` when they want to give an existing account access, but they could technically use both for both purposes. Not sure atm what's best but I feel like I personally would just want to know about one command first ("the intended way") and then possibly have a note somewhere where it says 'you could also use xxx', maybe including both here can cause some confusion / more questions.

valyo replied:
I believed that `dds project access grant --project "<Project ID>" --email "<email>"` fail if the specified email doesn't already have an account registered, but it seems I was wrong. Just checked it and it turns out it sends an invitation if there's no account. 
We can leave it as it is, with minor change in the text

aishling-scilifelab commented on docs/2_registration_project_access_session.md:
```diff
@@ -0,0 +1,142 @@
+
+The general upload command is [`dds data put`.](https://scilifelabdatacentre.github.io/dds_cli/data/#dds-data-put)
+~~~
+dds data put --project "<Project ID>" --source "<File or directory to upload>"
```
I think we should tell them about considering zipping before upload (because we want them to do that)

aishling-scilifelab commented on docs/2_registration_project_access_session.md:
```diff
@@ -0,0 +1,142 @@
+
+ - [ ] TASK: Release the project after making sure all the data was uploaded, remember to specify the correct project ID
+ - [ ] TASK: Download the project contents, and verify that they are the correct. You can download data with your unit admin account.
+ - [ ] Optional: Use another email you may have to invite yourself as a researcher, you don't need to register again.
```
I get what you mean here but for some reason find this sentence hard to parse. Maybe could say:
_If you have another account on Test DDS, you can invite that email as a researcher_
That said, 1) couldn't they just use plus addressing to instantly invite themselves as a researcher anyway? 2) how does it work when you want to set up authentication for 2 different accounts on the same computer? Presumably they'd have to end/delete the stored session token and then re-auth with the other account's credentials. Might be good to mention that

rv0lt replied:
It is just so they could test adding emails as a researcher, I don't intend on them to log in again. Because that could be too much mess.

aishling-scilifelab replied:
I still don't really understand this one.

valyo replied:
Yes, this is one of the places where the flow gets confusing to me. Maybe we should rearrange things, because the section title is "Release project and download data" but it includes giving access to researchers. Probably cleaner if we keep things to unit personnel here, i.e. release project and download data

i-oden replied:
Let me know if I'm misinterpreting etc.

But basically this is how it works:
- If they want to be able to download with a researcher account then they'd need to invite themselves as a researcher to that project, and then that researcher email would need to register an account. If that email has already been invited to the system and they have registered a researcher account previously, then they can invite that email to a project, and they then automatically have access to download, they wouldn't need to register again.
- In order to download with that researcher account, first they'd need to re-auth with the researcher credentials, and that would replace the existing local token (that belonged to the authenticated unit user account). 

So I think the "you don't need to register again" should be removed/moved somewhere else? If we want to have this optional suggestion here we could rephrase it to something like this, perhaps? (But we may want to include steps in that case)
> - [ ] Optional: If you have another email address, use your Unit Admin / Personnel account to invite that email as a researcher. Once you have registered the "researcher" account, authenticate using the researcher account credentials and use the `dds data get` command as described above to download the data.

valyo replied:
We should either change the text as Ina suggests, or don't mess things with Researcher account here. 
Maybe the best is to make this change now, and later on decide if it should go to another section or removed all together.

aishling-scilifelab reviewed (changes requested):
This looks good!

Unless we're covering it elsewhere, I think we should demonstrate the expected behaviour when both an upload and download are interrupted.

valyo commented on README.md:
```diff
@@ -4,3 +4,9 @@ This repository contains the instructions for the hands-on sessions during the w
 with SciLifeLab Unit Users (Autumn 2024).
```
```suggestion
for SciLifeLab Unit Users (Autumn 2024).
```

valyo commented on docs/1_installation.md:
```diff
@@ -0,0 +1,36 @@
+~~~
+
+![enter image description here](https://scilifelabdatacentre.github.io/dds_cli/_images/dds-version.svg)
+## Point to testing
```
```suggestion
## Using the testing instance
```

valyo commented on docs/1_installation.md:
```diff
@@ -0,0 +1,36 @@
+
+When you run any command, you can notice that, on top of the output, there is a URL, which by default is 
+
+> `https://delivery.scilifelab.se`
```
```suggestion
By default the client is using the production instance with URL

> `https://delivery.scilifelab.se`

You can see that URL in the output of any command you run.

```

rv0lt committed:
typos and bettered explanation

valyo commented on docs/2_registration_project_access_session.md:
```diff
@@ -0,0 +1,147 @@
+dds data put --project "<Project ID>" --source "<File or directory to upload>"
+~~~
+
+> NOTE: As of today, DDS can break when there are heavy uploads. We are working on fixing it, but on the meantime, we recommned to ZIP files when the uploads are big (Hundreds of files).
```
```suggestion
> NOTE: As of today, DDS has performance issues and upload failures when uploading many small files. We are working on fixing it, but on the meantime, we recommend you to archive (with ZIP or RAR, no compression required) files when the delivered data consists of hundreds or thousands of small files).
```

rv0lt committed:
notes for some commands

rv0lt committed:
bettered explanations

valyo commented on docs/2_registration_project_access_session.md:
```diff
@@ -0,0 +1,162 @@
+dds project access grant --project "<Project ID>" --email "<email>"
+~~~
+
+or
```
```suggestion
for giving access to an already existing account, or
```

aishling-scilifelab replied:
@valyo - given this suggestion, would you disagree with the text (see discussion immediately above this one) that says the commands are the same? 
Edit: I can see from below that yes you disagree there is no difference.

valyo commented on docs/2_registration_project_access_session.md:
```diff
@@ -0,0 +1,162 @@
+~~~
+dds user add [Email address] --role "Researcher" --project "<Project ID>"
+~~~
+
```
```suggestion

for sending an invitation for a new account, which will automatically obtain access to that specific project.

```

valyo commented on docs/2_registration_project_access_session.md:
```diff
@@ -0,0 +1,162 @@
+dds user add [Email address] --role "Researcher" --project "<Project ID>"
+~~~
+
+>There is no practical difference between this two command in this use case. However, as mentioned above, it is possible to automatically invite researchers when creating  a project. \
```
```suggestion
>As mentioned above, it is possible to automatically invite researchers when creating  a project. \
```

valyo reviewed (changes requested):
Let's have all the suggestion applied and can have another iteration.

rv0lt committed:
added spf explanation

valyo commented on docs/2_registration_project_access_session.md:
```diff
@@ -0,0 +1,195 @@
+dds data put --project "<Project ID>" --source "<File or directory to upload>"
+~~~
+
+> There are a number of other optional flags, which cab be found on the documentation link just above.
```
```suggestion
> There are a number of other optional flags, which can be found on the documentation link just above.
```

valyo commented on docs/2_registration_project_access_session.md:
```diff
@@ -0,0 +1,162 @@
+dds project access grant --project "<Project ID>" --email "<email>"
+~~~
+
+or
```
The end result is the same in terms that a researcher account gets access to a project, but the commands are for different scenarios. So I wouldn't say the commands are the same. Maybe we don't even need to say explicitly that they are the same or different. I will thy to find a solution

aishling-scilifelab committed:
Updates to installation/registration session

- Add extra detail to the README
- Make edits to the text as per code review comments

aishling-scilifelab commented:
@valyo - I have left 2 things unresolved: 1) Installation on Fedora and 2) have left Alvaro's text stating there is no difference between those 2 commands (because Ina's going to double check that later in the week).

valyo committed:
moved "Invite other Unit Staff" before "Create project"

valyo commented on docs/2_registration_project_access_session.md:
```diff
@@ -0,0 +1,200 @@
+~~~
+
+> There is no practical difference between these two commands for this use case. However, as mentioned above, it is 
+> possible to automatically invite researchers when creating a project. `dds project create (...) --researcher [Email address]`
```
```suggestion
> There is no practical difference between these two commands for this use case. As mentioned above, it is 
> also possible to automatically invite researchers when creating a project. `dds project create (...) --researcher [Email address]`
```

valyo commented on docs/2_registration_project_access_session.md:
```diff
@@ -0,0 +1,200 @@
+
+ - [ ] TASK: Release the project after making sure all the data was uploaded, remember to specify the correct project ID
+ - [ ] TASK: Download the project contents, and verify that they are the correct. You can download data with your unit admin account.
+ - [ ] Optional: Use another email you may have to invite yourself as a researcher, you don't need to register again.
```
```suggestion
 - [ ] Optional: If you have another email address, use your Unit Admin / Personnel account to invite that email as a researcher. Once you have registered the "researcher" account, authenticate using the researcher account credentials and use the dds data get command as described above to download the data.
```

valyo commented on docs/2_registration_project_access_session.md:
```diff
@@ -0,0 +1,200 @@
+
+~~~
+dds data get --source "<1st file or directory>" --source "<2nd file or directory>" [... etc] --project "<Project ID>"
+~~~
```
This one should be in the next section, IMO. 
Since it is too messy to suggest the change here, I will do it locally and commit it

valyo committed:
rearrange the download section

aishling-scilifelab commented:
@valyo can we get this one merged in, even if it's not perfect? Now that Alvaro is back especially it's easier if we can make little branches off main

valyo reviewed (changes requested):
Let's get all suggestions applied or denied, so that we can have clean view of the current material.

aishling-scilifelab commented:
@valyo can you apply those requested changes (I think you have the clearest picture of that ATM)?

valyo committed:
Update docs/2_registration_project_access_session.md

typo

valyo committed:
Update docs/2_registration_project_access_session.md

clarification

valyo committed:
Update docs/2_registration_project_access_session.md

reformulate

valyo commented:
@aishling-scilifelab I applied the changes and dismissed my changes request, now it's yours remaining and we can merge it

valyo committed:
rearrangement and edits in 2. Registration and project access

valyo commented:
I committed some changes that I already had, but it was probably stupid since the PR already has approvals. Please, take a look and merge, and then wi can continue in new PRs
